src/go/agwd/server/server.go

```diff
@@ -107,7 +110,8 @@ func startSctpdDownlinkServer(
 			target.Endpoint))
 	}
 
-	grpcServer := grpc.NewServer()
+	mw := middleware.NewMiddleware(cfgr.Config(), buf)
```

Comment:
could the middleware be passed in?

i.e. this func could take a `opt ...grpc.ServerOption` param and pass it along to `grpc.NewServer`

Reply:
yeah thatd actually be safer to make sure ppl use the same buffer